c7n/resources/appmesh.py

```diff
@@ -0,0 +1,260 @@
+        # the test data JSON files in the "placebo" directory.
+        arn = "arn"
+
+        # enum_spec : Defines the boto3 call used to find at least basic
+        # details all resources of the relevant type.  the data per
+        # resource can be further enriched by a detail_spec function.
+        # enum_spec is also used when we've received an event in which
+        # case the results from enum_spec are filtered to include only
+        # those in the event.
+        #
+        # If the enum function chosen allows a filter param to be
+        # specified then the filtering can be done on the server
+        # side. For instance, ASG uses "describe_auto_scaling_groups"
+        # as the enum function and "AutoScalingGroupNames" as a filter
+        # param to that function to limit the server side work.
+        # However, it seems that most "cloud custodian" integrations
+        # do not use this approach.  App mesh list_meshes doesn't
+        # support filtering.  ...
+        #
+        # https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/appmesh/client/list_meshes.html  # noqa
+        #
+        # and so when an event is received then the enum function gets
+        # called and the event id's get enriched.
+        #
+        # For example the specific identity found in an
+        # event. However, if the enum op doesn't support filtering
+        # then what will happen with events instead is a full list of
+        # resources followed by client side filtering.
+        #
+        # params:
+        #  enum_op - the aws api op
+        #  path - JMESPATH path to the field in the response that is the collection of result
+        #         objects
+        #  extra_args - eg {'maxResults': 100}
+        #
+        enum_spec = ('list_meshes', 'meshes', None)
+
+        # detail_spec: In many cases the enum_spec function is one of the
+        # "describe_" style functions that return a full'ish spec that
+        # is sufficient for the user detection, however in those cases
+        # where the enum_spec is a "list_" style function then the
+        # response to then enum call will tend to be lacking in detail and
+        # might just be a list of id's. In these cases it is generally
+        # necessary to define a "detail_spec" which can be used to
+        # enrich the values provided by the enum_spec.
+        #
+        # detail_op = boto api call name
+        # param_name = name of argument to boto api call
+        # param_key = name of field in enum_spec response to drive this call
+        # detail_path = path to pull out of the boto response and
+        #               return as the detail result if not provided
+        #               then whole response is included in results
+        detail_spec = ('describe_mesh', 'meshName', 'meshName', None)
+
+
+class DescribeGatewayDefinition(ChildDescribeSource):
+    def __init__(self, manager):
+        super().__init__(manager)

```

Comment:
fwiw, override isn't needed, describe virtual gateway includes the mesh name / parent id.